### ISSUE TITLE
Override Channels with `conda env create`

### ISSUE THREAD

**Issue by mfansler** [conda/conda#10498]
`conda env create` could benefit from an `--override-channels` flag similar to the `conda (create|install|update)` commands. This would help YAMLs more reliably reproduce an environment as specified. Otherwise, users must manipulate their configurations in order to match the intent of the YAML (i.e., set strict channel priorities or temporarily clear their global channel configuration).

**Comment by jamesmyatt** [comment]
I was surprised to find option this was not available. Seems mandatory in order to make `conda env export` work as intended.

**Comment by mforbes** [comment]
This is a crucial issue in terms of reproducible workflow.  I am brining it again so that the aggressive issue bots don't lock it for inactivity.

**Comment by n-a-sz** [comment]
Indeed, this is really annoying that it is recommenced to use this export format (https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#exporting-an-environment-file-across-platforms) but then when you would like to use it `conda env create` has less than half of the parameters compared to `conda create`.

**Comment by mfansler** [comment]
This feature is not implemented and is still desired.

To be formal, here's the current behavior:

```bash
$ conda env create --override-channels -f foo.yaml
usage: conda-env [-h] command ...
conda-env: error: unrecognized arguments: --override-channels
```

I should be noted that one can use the magic word `nodefaults` as a channel in the YAML to achieve the behavior. E.g.,

**foo.yaml**
```yaml
name: foo
channels:
  - conda-forge
  - bioconda
  - nodefaults
dependencies:
  - r-base=4.3
  - bioconductor-singlecellexperiment
```

**Comment by cohml** [comment]
@beeankha I would be willing to take this on. Could you please:

1. Assign it to me.
2. Provide any pointers you can think of to get me started as I navigate the source code?

Thanks.

**Comment by beeankha** [comment]
Hi @cohml , thanks for being willing to take on this work! Since the needs and requirements for the "collapse/alias `conda env` into `conda` commands" might have changed since this issue was added to [the related epic](https://github.com/conda/conda/issues/11633), please hang tight while I bring this up in a discussion with my teammates (we'll probably get a chance to discuss by next week). I will get back to you with more info after I find out more details.

**Comment by mfansler** [comment]
Not stale.
